### PR TITLE
feat: support .tar.zst bundles in wasm and more

### PR DESCRIPTION
**Pull request by LK**
also: fix stdlib embedding in wasm, fix symbol-relative path resolution, add publish time to `pcb info`

---

<a href="https://app.devin.ai/review/diodeinc/pcb/pull/623" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes WASM bundle ingestion/path normalization and workspace version enrichment from git tags, which can affect module resolution and `pcb info` outputs across environments.
> 
> **Overview**
> **WASM evaluation now accepts multiple bundle formats.** `pcb-zen-wasm` replaces the zip-backed `ZipFileProvider` with a `BundleFileProvider` that can read plain source zips, release-style bundles (extracting `src/`), and canonical `.tar.zst` bundles (zstd-decoded tar), including optional `metadata.json`-based main-file hints and improved path normalization/canonicalization.
> 
> **Workspace/package metadata is enriched with publish timestamps.** `MemberPackage` gains `published_at`, native workspace enrichment now records ISO-8601 tag timestamps for the latest package tag, and `pcb info -f json` is updated/covered by a new snapshot test to include this field.
> 
> **Diagnostics and resolution edge cases are tightened.** Prelude injection failures now emit explicit load diagnostics (instead of silently skipping), `format_package_uri` avoids cache-path rewriting when `cache_dir` is unset, and test harness resolution/stdlib fixtures were updated to include default KiCad asset dependency mappings.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8ff05824a5c90eae39fed13bd49bd3c0e55fa93a. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: crates/pcb-zen-wasm/src/lib.rs [hunk 1]
```diff
@@ -285,10 +293,134 @@ impl FileProvider for ZipFileProvider {
...
+            format!("Failed to parse bundle as zip ({zip_err}) or .tar.zst ({tar_err})")
+        }),
+    }
+}
+
+fn parse_zip_bundle(bundle_bytes: &[u8]) -> Result<ParsedBundle, zip::result::ZipError> {
+    let mut archive = ZipArchive::new(Cursor::new(bundle_bytes))?;
+    let is_release_bundle = archive.by_name("metadata.json").is_ok();
+    let hinted_main_file = if is_release_bundle {
+        archive.by_name("metadata.json").ok().and_then(|mut file| {
```
Comment (akhilles) [inline crates/pcb-zen-wasm/src/lib.rs]: double lookup here. this is inside a conditional that asserts `archive.by_name("metadata.json").is_ok()`

File: crates/pcb-zen-wasm/src/lib.rs [hunk 2]
```diff
@@ -424,7 +652,15 @@ pub fn evaluate_impl(
         main_file.to_string()
     };
 
-    let main_path = PathBuf::from(&main_file);
+    let requested_main_path = PathBuf::from(&main_file);
+    let main_path = if requested_main_path.is_absolute() {
```
Comment (akhilles) [inline crates/pcb-zen-wasm/src/lib.rs]: curious why this conditional is needed. if abs or not, the behavior seems the same

**Review by akhilles — APPROVED** [review]

**Comment by LK** [comment]
@cursor push b186941452
